refactor: BTS-228 / 스터디룸 생성 플로우 리팩토링

JangHwanPark opened the pull request:
## ✅ 체크리스트
- [ ] 코드에 린트 에러가 없습니다.
- [ ] 코드가 올바르게 포맷팅되었습니다.
- [ ] 빌드가 올바르게 실행되었습니다.

## 📌 변경 사항
### invite-member 라우터 생성
<img width="1413" height="937" alt="image" src="https://github.com/user-attachments/assets/36f96c2f-579f-455f-9fd7-76d65e4bfbdb" />

- `/studyrooms/[id]/invite-member` 페이지 신규 생성  
- 스터디룸 생성 완료 후 초대 플로우로 자연스럽게 이동하도록 연결  
- 초대 필드, 유효성, 안내문, 버튼 등 기본 UI/UX 구성 완료 

### CreateStudyRoomFlow
- router.push → router.replace 변경 (히스토리 스택 최소화)
- 초대/대시보드 이동 로직을 handleSubmitCondition 으로 통합
- buildNextRoute() 유틸 함수 추가로 라우팅 관리
- RHF(useForm)의 handleSubmit 흐름을 유지하면서도 조건 기반 분기 적용
- dialogReducer 상태와 연동해 ConfirmDialog 제어 추가

### ConfirmDialog
<img width="768" height="463" alt="image" src="https://github.com/user-attachments/assets/15f36a16-9a75-40e4-bd07-bb582c168374" />

- 기존 단일 목적 다이얼로그를 공용 ConfirmDialog 형태로 1차 통합
- variant(ok / confirm / confirm-cancel / delete) 기반으로 재사용성 강화
- DialogAction 기반 dispatch 로직으로 상태 관리 통일
- studyroom(create) 전용으로 우선 적용 (다른 도메인은 추후 통합 예정)

### SEO 설정
- 홈(/) 페이지 메타데이터 추가
- robots.txt 자동 생성(Next.js MetadataRoute 기반)
- sitemap.xml 자동 생성(홈 경로 기준)
- changeFrequency, priority, lastModified 설정 추가
- SEO 설정은 현재 홈 페이지에만 적용됨
- 추후 스펙에 맞춰 확장 예정

## 🧐 관련 이슈
[[FE] 스터디룸 생성 화면 개발 v2](https://daechi-online.atlassian.net/jira/software/projects/BTS/boards/1?selectedIssue=BTS-228)

## TODO
- [x] 이 PR 내부에 메타데이터 설정(meta tags) 및 robots파일 설정을 추가한 후 병합 예정

JangHwanPark committed:
:recycle: refactor: BTS-228 / 스터디룸 생성페이지 내 사이드메뉴 컴포넌트 추가

JangHwanPark committed:
:construction: chore: BTS-228 / 스터디룸 초대페이지 임시 킵파일 추가

- 추후 대시보드(dashboard)내부로 리팩토링 예정

JangHwanPark committed:
:recycle: refactor: BTS-228 / 스터디룸 초대페이지 마크업 작업

JangHwanPark committed:
:recycle: refactor: BTS-228 / 스터디룸 생성 및 초대 로직 리팩토링

- InviteMemberPage에서 react-hook-form 제거 및 단순 onClick 기반 제출로 변경
- CreateStudyRoomFlow 내 router.push → router.replace로 변경하여 내비게이션 일관성 개선
- 초대 성공 후 대시보드 이동 로직 추가

JangHwanPark committed:
:recycle: refactor: BTS-228 / 스터디룸 생성 플로우 리팩토링 및 ConfirmDialog 1차 통합

CreateStudyRoomFlow
- router.push → router.replace 로 변경 (히스토리 스택 최소화)
- 초대/대시보드 이동 로직을 handleSubmitCondition 으로 통합
- buildNextRoute() 유틸 함수 추가로 라우팅 조건 분리

ConfirmDialog
- 기존 단일 목적 ConfirmDialog -> 공용 ConfirmDialog 형태로 1차 통합
- variant(ok, confirm, confirm-cancel, delete) 지원
- DialogAction 기반 dispatch 연결(CLOSE 자동 처리)

TODO(ConfirmDialog): 현재는 studyroom 전용(common 하위 컴포넌트)
- ConfirmDialog를 전역 공용 컴포넌트로 생성(/components/common/dialog)
- 추후 /components/dialog 단일 공용 다이얼로그로 분리 예정
- 입력 필드 포함 다이얼로그(InputDialog 등)는 예외적으로 별도 유지

chay140 reviewed (approved):
PR 내용 확인했습니다! 메타데이터와 robots 파일 올리시는대로 머지하면 좋을 것 같습니다!

JangHwanPark committed:
:card_file_box: chore: BTS-228 / SEO 설정 추가(robots + sitemap)

홈(/) 페이지 메타데이터 설정 추가
- OG 이미지 경로: /logo.svg
- SITE_URL: https://the-edu.vercel.app

robots.ts설정 추가(Next.js MetadataRoute 기반)
- Allow: '/' (홈만 인덱싱)
- Disallow: /login, /register, /studyrooms, /dashboard, /api, /_next
- Sitemap: https://the-edu.vercel.app/sitemap.xml

sitemap.ts 추가(Next.js MetadataRoute 기반 자동 생성)
- changeFrequency: weekly
- priority: 1
- lastModified: new Date()
- 홈 경로만 우선 포함

JangHwanPark commented:
배포 일정 관계로 이번에는 바로 병합을 진행하겠습니다.
추후 QA 도중 또는 운영 중 이슈 발생 시 크리티컬한 이슈 위주로 핫픽스 대응 예정입니다.
